src/renderers/dom/shared/eventPlugins/ChangeEventPlugin.js

```diff
@@ -258,6 +258,26 @@ function getTargetInstForInputOrChangeEvent(topLevelType, targetInst) {
   }
 }
 
+function handleControlledInputBlur(inst, node) {
+  // TODO: In IE, inst is occasionally null. Why?
+  if (inst == null) {
+    return;
+  }
+
+  // Fiber and ReactDOM keep wrapper state in separate places
+  let state = inst._wrapperState || node._wrapperState;
+
+  if (!state || !state.controlled || node.type !== 'number') {
+    return;
+  }
+
+  // If controlled, assign the value attribute to the current value on blur
+  let value = '' + node.value;
+  if (node.getAttribute('value') !== value) {
+    node.setAttribute('value', value);
```

Comment:
Assigning defaultValue has the effect of updating the value attribute. Which puts us in the same boat. the only trick with supporting form reset would be the execution order. Last I checked, reset emits _before_ the input actually changes. We'd have to do some sort of next tick behavior. I'm curious what the order of execution would look like to a user in the browser. Would it be transparent?

Something I can figure out.